crates/sui-graphql-rpc/src/example_driver.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::client::simple_client::SimpleClient;
+use crate::config::{ConnectionConfig, ServiceConfig};
+use crate::server::simple_server::start_example_server;
+use std::collections::BTreeMap;
+use std::{fs, io::Read, path::PathBuf};
+
+const QUERY_EXT: &str = "graphql";
+const EXPECTED_RESULT_EXT: &str = "exp";
+
+/// Loop through all files in the examples directory which end in .graphql and .exp
+/// and load the contents into strings.
+/// Then, run the queries and compare the results to the expected results.
+pub async fn verify_examples_impl() {
+    let mut buf: PathBuf = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    buf.push("examples");
+
+    let mut queries = BTreeMap::new();
+    let mut expected_results = BTreeMap::new();
+
+    let config = ConnectionConfig {
+        port: 8123,
+        ..Default::default()
+    };
+    let server_url = config.server_url();
+
+    tokio::spawn(async move {
+        start_example_server(config, ServiceConfig::default()).await;
+    });
```

Comment:
What does this do? Does this start some local instance with some data in it or does it reference some db that was initialized externally?

Reply:
looking at the code it seems this uses testnet as a data source:

```
rpc_url: "https://fullnode.testnet.sui.io:443/".to_string(),
```

I'd argue this is not a great idea and gets us into a situation where if testnet is down, or reset, or wiped, then all of these tests would start to fail causing us to be unable to land any changes easily (without first disabling them)